### PR TITLE
Add SkiaPerfPoint and FlutterEngineMetricPoint

### PR DESCRIPTION
**Pull request by liyuqian**
Unit tests that cover their translations are also added.

### CODE REVIEW

File: dev/benchmarks/metrics_center/lib/src/github_helper.dart [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+class GithubHelper {
+  /// Return the singleton helper.
+  factory GithubHelper() {
+    return _singleton;
+  }
+
+  GithubHelper._internal();
+
+  /// The result is cached in memory so querying the same thing again in the
+  /// same process is fast.
```
Comment (keyonghan) [inline dev/benchmarks/metrics_center/lib/src/github_helper.dart:21]: From the test, 1000 queries per second is needed. Shall the doc reflect that?

File: dev/benchmarks/metrics_center/test/skiaperf_test.dart [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+void main() {
+  const double kValue1 = 1.0;
+  const double kValue2 = 2.0;
+
+  const String kFrameworkRevision1 = '9011cece2595447eea5dd91adaa241c1c9ef9a33';
+  const String kTaskName = 'analyzer_benchmark';
+  const String kMetric1 = 'flutter_repo_batch_maximum';
+  const String kMetric2 = 'flutter_repo_watch_maximum';
+
+  final MetricPoint cocoonPointRev1Metric1 = MetricPoint(
```
Comment (keyonghan) [inline dev/benchmarks/metrics_center/test/skiaperf_test.dart:24]: How a cocoon Point is related to skiaperf point and flutterengine point?

File: dev/benchmarks/metrics_center/lib/src/flutter.dart [hunk 3]
```diff
@@ -0,0 +1,22 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:metrics_center/src/common.dart';
+
+/// Convenient class to capture the benchmarks in the Flutter engine repo.
+class FlutterEngineMetricPoint extends MetricPoint {
```
[the remark this answers is not in the document]
Comment (liyuqian) [reply dev/benchmarks/metrics_center/lib/src/flutter.dart:8]: I'm planning to add more Flutter-related but non-engine-related classes such as `FlutterDestination` to `flutter.dart`. I can also create a `flutter` directory and put `flutter_engine.dart` and other files inside it. For now, I don't see many files or content to be needed so I adopted a single `flutter.dart` for now. I'm happy to change either now or in the future if you still prefer it considering the upcoming changes.

File: dev/benchmarks/metrics_center/lib/src/github_helper.dart (hunk 1 above)
Comment (liyuqian) [reply dev/benchmarks/metrics_center/lib/src/github_helper.dart:21]: Sure, added.

File: dev/benchmarks/metrics_center/lib/src/skiaperf.dart [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+  // Equivalent to tags without git repo, git hash, and name because those two
+  // are already stored somewhere else.
+  final Map<String, String> _options;
+}
+
+const String kSkiaPerfGitHashKey = 'gitHash';
+const String kSkiaPerfResultsKey = 'results';
+const String kSkiaPerfValueKey = 'value';
+const String kSkiaPerfOptionsKey = 'options';
+
```
[the remark this answers is not in the document]
Comment (liyuqian) [reply dev/benchmarks/metrics_center/lib/src/skiaperf.dart:186]: Done.

File: dev/benchmarks/metrics_center/test/skiaperf_test.dart (hunk 2 above)
Comment (liyuqian) [reply dev/benchmarks/metrics_center/test/skiaperf_test.dart:24]: Cocoon points (framework or device lab perf metrics) are not related to FlutterEnginePoint. Cocoon points should be able to convert to SkiaPerfPoint. I'll probably add more documentations when I migrate CocoonSource / BenchmarkMetricPoint in the future.

File: dev/benchmarks/metrics_center/lib/src/flutter.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (keyonghan) [reply dev/benchmarks/metrics_center/lib/src/flutter.dart:8]: Thanks for the explanation. Then it should be fine to use flutter.

**Review by keyonghan — APPROVED** [review]
LGTM

**Pull request merged**